client/src/components/CodeUploadDialog/CodeUploadDialog.jsx

```diff
@@ -0,0 +1,144 @@
+import React, { useState } from 'react';
+import { 
+  Dialog,
+  Box,
+  Grid,
+  TextField,
+  MenuItem,
+  useMediaQuery,
+} from '@material-ui/core';
+import { useTheme } from '@material-ui/core/styles';
+
+import useStyles from './CodeUploadDialog.css';
+import CodeEditor from '../CodeEditor';
+import SubmitButton from '../ui/SubmitButton';
+
+const CodeUploadDialog = ({ open, onClose }) => {
+  const theme = useTheme();
+  const classes = useStyles();
+
+  const languages = {
+    python: 'Python',
+    java: 'Java',
+    c_cpp: 'C++',
+    javascript: 'JavaScript',
+    ruby: 'Ruby',
+  };
+  const defaultLanguage = Object.keys(languages)[0];
+
+  const [ title, setTitle ] = useState('');
+  const [ language, setLanguage ] = useState(defaultLanguage);
+  const [ codeSnippet, setCodeSnippet ] = useState('');
+  const [ comments, setComments ] = useState('');
+
+  const handleTitleChange = event => {
+    setTitle(event.target.value);
+  };
+
+  const handleLanguageChange = event => {
+    setLanguage(event.target.value);
+  };
+
+  const handleCodeSnippetChange = value => {
+    setCodeSnippet(value);
+  };
+
+  const handleCommentsChange = event => {
+    setComments(event.target.value);
+  };
```

Comment:
You can probably combine handleCommentsChange, handleTitleChange, and handleLanguageChange into one function and use the event.target.name field and a switch statement 

Add a name to the TextFields to know which state to change